networkmanager/traffic.go

```diff
@@ -68,28 +69,60 @@ type trafficData struct {
 }
 
 type trafficMonitoring struct {
-	iptables         *iptables.IPTables
-	trafficPeriod    int
-	skipAddresses    string
-	inChain          string
-	outChain         string
-	trafficMap       map[string]*trafficData
-	serviceChainsMap map[string]*trafficChains
-	trafficStorage   TrafficStorage
+	iptables          IPTablesInterface
+	trafficPeriod     int
+	skipAddresses     string
+	inChain           string
+	outChain          string
+	trafficMap        map[string]*trafficData
+	instanceChainsMap map[string]*trafficChains
+	trafficStorage    TrafficStorage
 }
 
+type IPTablesInterface interface {
+	ListWithCounters(table, chain string) ([]string, error)
+	Append(table, chain string, rulespec ...string) error
+	Delete(table, chain string, rulespec ...string) error
+	NewChain(table, chain string) error
+	Insert(table, chain string, pos int, rulespec ...string) error
+	ClearChain(table, chain string) error
+	DeleteChain(table, chain string) error
+	ListChains(table string) ([]string, error)
+}
+
+/***********************************************************************************************************************
+ * Vars
+ **********************************************************************************************************************/
+
+// ErrNotExist not exist entry error.
+var (
+	ErrNotExist     = errors.New("entry does not exist")
+	ErrRuleNotExist = errors.New("chain rule not exist")
+)
+
+var IsSamePeriod func(t1, t2 time.Time) (result bool) // nolint:gochecknoglobals
```

Comment:
Why do you need to mock this function? It can be covered with a unit test.

Reply:
left the IsSamePeriod function but initialized when declared